### ISSUE TITLE
Doesn't work on Metro Exodus Enhanced Edition

### ISSUE THREAD

**Issue by mufaaf** [Merrit/nyrna#228]
Hey folks. First of all, Nyrna is an amazing software. Completely changed how I stream games to my Steam Deck OLED!

I have used it for several games and have had no issues. However, with Metro Exodus Enhanced Edition, it doesn't work at all. I assign hotkey to enable/disable suspend feature but with Metro it doesn't register. I even changed the hotkey and even assigned app-specific hotkey but nothing works.

Any idea how I can make it work? Thanks!

Edit: I am on Windows 10.

**Comment by Merrit** [comment]
Please try running Nyrna as admin

**Comment by mufaaf** [comment]
> Please try running Nyrna as admin

Sadly, didn't work.

**Comment by Merrit** [comment]
Can you use Nyrna if you Alt+Tab out of the game and use the GUI to suspend/resume?

**Comment by mufaaf** [comment]
> Can you use Nyrna if you Alt+Tab out of the game and use the GUI to suspend/resume?

Yup, that works just fine. But it doesn't work with shortcuts.

**Comment by Merrit** [comment]
Seems to be one of those games that grabs exclusive control of the keyboard, not allowing other apps to use shortcuts.

We can't do anything about that unfortunately, but I added Metro Exodus to the troubleshooting section of the docs: https://nyrna.merritt.codes/docs/troubleshooting/#metro-exodus

**Comment by vlad54rus** [comment]
A possible workaround is to use patched dinput8.dll.
I haven't played Exodus, but I have played their previous titles - Metro Redux, and they had the same issue. These games call DirectInput SetCooperativeLevel function with DISCL_EXCLUSIVE flag set. What I did was:
1. Copied 64-bit dinput8.dll from C:\Windows\System32 folder into the game folder.
2. Opened it in IDA, made sure it downloads dinput8 debug symbols from the Microsoft servers.
3. Found a function named CKbd_SetCooperativeLevel, looked up which part of it checks for the DISCL_EXCLUSIVE (0x01) flag and replaced it with NOPs.
